src/main/groovy/com/vherasymenko/avro/decoder/core/AvroDecoderService.groovy

```diff
@@ -0,0 +1,25 @@
+package com.vherasymenko.avro.decoder.core
+
+import event.course_install.CourseInstall
+import org.apache.avro.generic.GenericDatumReader
+import org.apache.avro.generic.GenericRecord
+import org.apache.avro.io.DatumReader
+import org.apache.avro.io.DecoderFactory
+
+/**
+ * The avro decoder.
+ */
+class AvroDecoderService implements AvroDecoderPort {
+
+    @Override
+    void decodeEvent( String event ) {
+        def inputStream = new ByteArrayInputStream( event.bytes )
```

Comment:
Be careful. I don't believe you can be guaranteed that you are getting UTF-8 encoded bytes.  You might consider using the explicit `event.getBytes( 'UTF-8' )`.  At the very least, double check Groovy's API documentation and see what encoding is used when getting bytes from a string.